Unity-Technologies/ScriptableRenderPipeline
# Improve play mode change speed

### Purpose of this PR
Improved the speed of HDRP when entering / exiting play mode with the experimental play mode option enabled by a factor of ~25.

- Changed the texture caches to use a RenderTexture instead of a standard Texture which avoid to pay the cost of a unnecessary texture upload (that was taking almost one second).
- Also changed some TextureFormat ot GraphicsFormat because it can be used on both Texture and RenderTextures.

**Side note:** HDRP is still the slowest part of the enter play mode thing, but now it takes something like 50ms instead 1200 which is more reasonable. The rest ofd the time is spread over a lot of things:
![image](https://user-images.githubusercontent.com/6877923/71727438-830efd80-2e3a-11ea-971f-e3e1b37cbeca.png)

### QA
Focus on testing Cookies, reflection probes and planars, the behavior must not have changed. You can also ensure that there is a performance gain when we change a setting in the HDRP asset, or when we change the play mode.
